### PR TITLE
Sync settings

### PR DESCRIPTION
**Pull request by monusingh-1**
### Description
Sync settings when trying to update leader index metadata.
 
### Issues Resolved
https://github.com/opensearch-project/cross-cluster-replication/issues/992
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt [hunk 1]
```diff
@@ -26,14 +26,20 @@ import kotlinx.coroutines.asCoroutineDispatcher
 import kotlinx.coroutines.launch
 import org.apache.logging.log4j.LogManager
 import org.opensearch.action.ActionListener
+import org.opensearch.action.admin.indices.alias.IndicesAliasesRequest
```
Comment (monusingh-1) [inline src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt:29]: Will remove unused imports in next commit.

Comment (sricharanvuppu) [reply src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt:29]: +1

File: src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt [hunk 2]
```diff
@@ -166,6 +178,120 @@ class TransportReplayChangesAction @Inject constructor(settings: Settings, trans
         return WriteReplicaResult(request, location, null, replicaShard, log)
     }
 
+    // fetches the index settings from the leader cluster and applies it to the local cluster's index settings.
+    private suspend fun syncRemoteSettings(leaderAlias: String, leaderIndex: String, followerIndex: String) {
```
Comment (ankitkala) [inline src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt:182]: Like discussed offline, let's use common logic between replay and index replication task for mapping and settings sync.

Comment (monusingh-1) [reply src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt:182]: ack

Comment (soosinha) [inline src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt:182]: This function is being added mainly to be able to update mappings which use analyzer defined in the settings.
We can reduce the scope of this function to just check if there is any change in the `index.analysis.*` settings and then update that. This way we can remove most of the logic of creating the settings and closing the index.

File: src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt [hunk 3]
```diff
@@ -124,6 +131,11 @@ class TransportReplayChangesAction @Inject constructor(settings: Settings, trans
             var result = primaryShard.applyTranslogOperation(op, Engine.Operation.Origin.PRIMARY)
             if (shouldSyncMappingAndRetry(result)) {
                 waitForMappingUpdate {
+                     // fetch index settings from the leader cluster when applying on PRIMARY...
+                    try{
```
Comment (soosinha) [inline src/main/kotlin/org/opensearch/replication/action/replay/TransportReplayChangesAction.kt:135]: We should not catch the exception

**Pull request closed without merging**